SEFramework/SEFramework/Convolution/DFT.h

```diff
@@ -218,23 +218,19 @@ class DFTConvolution {
   }
 
 protected:
-  void padKernel(int width, int height, typename std::vector<T>::iterator out) const {
-    auto padded = PaddedImage<T>::create(m_kernel, width, height);
-    auto center = PixelCoordinate{width / 2, height / 2};
-    if (width % 2 == 0) center.m_x--;
-    if (height % 2 == 0) center.m_y--;
+  void padKernel(ConvolutionContext& context) const {
+    auto padded = PaddedImage<T>::create(m_kernel, context.m_padded_width, context.m_padded_height);
+    auto center = PixelCoordinate{context.m_padded_width / 2, context.m_padded_height / 2};
+    if (context.m_padded_width % 2 == 0) center.m_x--;
+    if (context.m_padded_height % 2 == 0) center.m_y--;
     auto recenter = RecenterImage<T>::create(padded, center);
 
-    dumpImage(recenter, out);
+    dumpImage(recenter, context.m_kernel_transform);
   }
 
-  void dumpImage(const std::shared_ptr<const Image<T>> &img, typename std::vector<T>::iterator out) const {
-    auto chunk = img->getChunk(0, 0, img->getWidth(), img->getHeight());
-    for (int y = 0; y < chunk->getHeight(); ++y) {
-      for (int x = 0; x < chunk->getWidth(); ++x) {
-        *out++ = chunk->getValue(x, y);
-      }
-    }
+  void dumpImage(const std::shared_ptr<const Image<T>> &img, std::vector<T>& work_area) const {
+    const auto chunk = img->getChunk(0, 0, img->getWidth(), img->getHeight());
```

Comment:
There is probably some gains if we add a `getChunk`-like call where the outermost caller allocates the memory.
Here we are allocating two buffers of the same size only to copy between them.